src/main/java/com/github/ibm/mapepire/requests/PreparedExecute.java

```diff
@@ -23,36 +24,31 @@ public PreparedExecute(final DataStreamProcessor _io, final JsonObject _reqObj,
 
     @Override
     protected void go() throws Exception {
-        JsonElement parms = super.getRequestField("parameters");
-        boolean isBatch = super.getRequestFieldBoolean("batch", false);
+        JsonArray parms = super.getRequestField("parameters").getAsJsonArray();
+        boolean isBatch = !parms.isEmpty() && parms.get(0).isJsonArray();
+        boolean hasResultSet = false;
+        int batchUpdateCount = 0;
+
         PreparedStatement stmt = m_prev.getStatement();
         if (isBatch) {
-            if (null == parms) {
-                stmt.executeLargeBatch();
-                addReplyData("update_count", stmt.getLargeUpdateCount());
-                return;
-            }
             JsonArray arr = parms.getAsJsonArray();
-            JsonElement firstElement = arr.get(0);
             int batch_ops_added = 0;
-            if (firstElement.isJsonArray()) {
-                for (int i = 1; i <= arr.size(); ++i) {
-                    addJsonArrayParameters(stmt, arr.get(-1 + i).getAsJsonArray());
-                    m_prev.getStatement().addBatch();
-                }
-                batch_ops_added += arr.size();
-            } else {
-                addJsonArrayParameters(stmt, arr);
+            for (int i = 0; i < arr.size(); i++) {
+                addJsonArrayParameters(stmt, arr.get(i).getAsJsonArray());
                 m_prev.getStatement().addBatch();
-                batch_ops_added++;
             }
+            batch_ops_added += arr.size();
             addReplyData("batch_added", batch_ops_added);
-            return;
-        } else if (null != parms) {
-            addJsonArrayParameters(stmt, parms.getAsJsonArray());
+            int updateCount[] = stmt.executeBatch();
+            batchUpdateCount = Arrays.stream(updateCount).sum();
+        } else{
+            if (parms != null) {
+                addJsonArrayParameters(stmt, parms.getAsJsonArray());
+            }
+            hasResultSet = stmt.execute();
         }
-        //TODO: (here and other places) handle multiple returned results
-        if (stmt.execute()) {
+
+        if (hasResultSet) {
```

Comment:
If `isBatch` is true, looks like `hasResultSet` is always false. Is this expected?

Reply:
yes